src/ui/components/AbstractToolbar.js

```diff
@@ -1,31 +1,381 @@
-import EquipmentToolbar from "../pages/equipment/components/EquipmentToolbar";
-import WorkorderToolbar from "../pages/work/WorkorderToolbar";
-import PartToolbar from "../pages/part/PartToolbar";
-import LocationToolbar from "../pages/equipment/location/components/LocationToolbar";
 import React from 'react';
+import MenuItem from "@material-ui/core/MenuItem";
+import Tooltip from "@material-ui/core/Tooltip";
+import IconButton from "@material-ui/core/IconButton";
+import Divider from '@material-ui/core/Divider';
+import { WorkorderIcon } from "eam-components/dist/ui/components/icons";
+import OpenInNewIcon from 'mdi-material-ui/OpenInNew'
+import {Barcode, ContentCopy, EmailOutline, Map, Printer, Domain, Camera} from 'mdi-material-ui';
+import { RadiationIcon } from "eam-components/dist/ui/components/icons";
+import { Link } from "react-router-dom";
 
-export const TOOLBARS = {
+export const ENTITY_TYPE = {
     WORKORDER: 'WORKORDER',
     EQUIPMENT: 'EQUIPMENT',
     PART: 'PART',
-    LOCATION: 'LOCATION'
+    LOCATION: 'LOCATION',
+}
+
+const VIEW_MODES = {
+    MENU_ITEMS: "MENUITEMS",
+    TOOLBAR_ICONS: "TOOLBARICONS",
+}
+
+export const BUTTON_KEYS = {
+    COPY: "COPY",
+    EMAIL: "EMAIL",
+    SHOW_ON_MAP: "SHOW_ON_MAP",
+    NEW_ENTITY: "NEW_ENTITY",
+    LINK_TO_WORKORDER: "LINK_TO_WORKORDER",
+    SHOW_IN_INFOR: "SHOW_IN_INFOR",
+    PRINT: "PRINT",
+    BARCODING: "BARCODING",
+    OSVC: "OSVC",
+    DISMAC: "DISMAC",
+    TREC: "TREC",
+    CREATE_WORKORDER: "CREATE_WORKORDER",
 }
 
 class AbstractToolbar extends React.Component {
 
-    toolbarComponents = {
-        [TOOLBARS.WORKORDER]: WorkorderToolbar,
-        [TOOLBARS.EQUIPMENT]: EquipmentToolbar,
-        [TOOLBARS.PART]: PartToolbar,
-        [TOOLBARS.LOCATION]: LocationToolbar
+    iconStyle = {
+        width: 20,
+        height: 20
+    };
+
+    iconMenuStyle = {
+        marginRight: 5,
+        width: 20,
+        height: 20
+    };
+
+    toolbarIconsStyle = {
+        alignItems: "center",
+        display: "flex"
+    }
+    
+    verticalLineStyle = {
+        height: 25,
+        borderRight: "1px solid gray",
+        margin: 5
     }
 
-    render() {
-        const { _toolbarType} = this.props;
-        const Toolbar = this.toolbarComponents[_toolbarType];
-        if(!Toolbar) return null;
+    linkStyle = {
+        display: "flex",
+        color: "black",
+        alignItems: "center",
+        textDecoration: "none"
+    }
+
+    getButtonDefinitions = () => {
+        return {
+            [BUTTON_KEYS.COPY] : {
+                isVisible: () => true,
```

Comment:
Could buttons being visible by default be a good idea, to save some typing/reading?